user_guide_src/source/libraries/publisher.rst

```diff
@@ -0,0 +1,442 @@
+
+Automation and Discovery
+========================
+
+You may have regular publication tasks embedded as part of your application deployment or upkeep. ``Publisher`` leverages
+the powerful ``Autoloader`` to locate any child classes primed for publication::
+
+	use CodeIgniter\CLI\CLI;
+	use CodeIgniter\Publisher\Publisher;
+	
+	foreach (Publisher::discover() as $publisher)
+	{
+		$result = $publisher->publish();
+
+		if ($result === false)
+		{
+			CLI::error(get_class($publisher) . ' failed to publish!', 'red');
+		}
+	}
+
+By default ``discover()`` will search for the "Publishers" directory across all namespaces, but you may specify a
+different directory and it will return any child classes found::
+
+	$memePublishers = Publisher::discover('CatGIFs');
+
+Most of the time you will not need to handle your own discovery, just use the provided "publish" command::
+
+	> php spark publish
+
+By default on your class extension ``publish()`` will add all files from your ``$source`` and merge them
+out to your destination, overwriting on collision.
+
+********
+Examples
+********
+
+Here are a handful of example use cases and their implementations to help you get started publishing.
+
+File Sync Example
+=================
+
+You want to display a "photo of the day" image on your homepage. You have a feed for daily photos but you
+need to get the actual file into a browsable location in your project at **public/images/daily_photo.jpg**.
+You can set up :doc:`Custom Command </cli/cli_commands>` to run daily that will handle this for you::
+
+	<?php
+
+	namespace App\Commands;
+
+	use CodeIgniter\CLI\BaseCommand;
+	use CodeIgniter\Publisher\Publisher;
+	use Throwable;
+
+	class DailyPhoto extends BaseCommand
+	{
+		protected $group       = 'Publication';
+		protected $name        = 'publish:daily';
+		protected $description = 'Publishes the latest daily photo to the homepage.';
+
+		public function run(array $params)
+		{
+			$publisher = new Publisher('/path/to/photos/', FCPATH . 'assets/images');
+
+			try
+			{
+				$publisher->addPath('daily_photo.jpg')->copy(true); // `true` to enable overwrites
+			}
+			catch (Throwable $e)
+			{
+				$this->showError($e);
+			}
+		}
+	}
+
+Now running ``spark publish:daily`` will keep your homepage's image up-to-date. What if the photo is
+coming from an external API? You can use ``addUri()`` in place of ``addPath()`` to download the remote
+resource and publish it out instead::
+
+	$publisher->addUri('https://example.com/feeds/daily_photo.jpg')->copy(true);
+
+Asset Dependencies Example
+==========================
+
+You want to integrate the frontend library "Bootstrap" into your project, but the frequent updates makes it a hassle
+to keep up with. You can create a publication definition in your project to sync frontend assets by adding extending
+``Publisher`` in your project. So **app/Publishers/BootstrapPublisher.php** might look like this ::
```

Comment:
Can you try removing the space between `::`
```suggestion
``Publisher`` in your project. So **app/Publishers/BootstrapPublisher.php** might look like this::
```

Reply:
I just added that this last round.

Reply:
I think I might see it, missing a single quote